cmd/podman/images/scp.go

```diff
@@ -0,0 +1,317 @@
+	} // else we are loading from the ssh client
+	podman := os.Args[0]
+	if v, found := os.LookupEnv("PODMAN_BINARY"); found {
+		podman = v // get path to podman bin
+	}
+	run := podman + " image save " + imageFileOut + " --output=" + imageFileIn // run ssh image load of the file copied via scp. Files are reverse in thie case...
+	var buffer bytes.Buffer
+	session.Stdout = &buffer
+
+	if err := session.Run(run); err != nil { // run the command declared above
+		return "", err
+	}
+	n, err := scpD.CopyFrom(dial, imageFileIn, imageFileIn)
+	if err != nil {
+		fmt.Print(n, " bytes copied before err")
+		return "", err
+	}
+	return "", nil
+}
+
+// parseLocation returns the valid connection data based off of the information provided by the user
+// iden is a string array of ssh identity keys, url is a string array of paths to ssh clients. and serv is
+// an array of service destinations
+func parseLocation(args []string, cfg *config.Config) ([]string, []string, map[string]config.Destination) {
+	iden := []string{}
+	url := []string{}
+	serv := map[string]config.Destination{}
+	if strings.Contains(args[0], "::") {
+		if len(args) == 2 { // if we have a connection name specified
+			if !(strings.Contains(args[1], "::")) { // if an image is specified, this mean we are loading from our client
+				scpOpts.Connections = append(scpOpts.Connections, args[0])
+				scpOpts.ToRemote = true
+			} else if strings.Contains(args[1], "::") { // both remote clients
+				scpOpts.FromRemote = true
+				scpOpts.ToRemote = true
+				if len(strings.Split(args[0], "::")) == 1 || (len(strings.Split(args[0], "::")) == 2 && len(strings.Split(args[1], "::")) == 2) { // is save->load w/ one image name or save-load w/ two image names
+					scpOpts.Connections = append(scpOpts.Connections, args[0])
+					scpOpts.Connections = append(scpOpts.Connections, args[1])
+				} else { // else its a load save (order of args)
+					scpOpts.Connections = append(scpOpts.Connections, args[1])
+					scpOpts.Connections = append(scpOpts.Connections, args[0])
+				}
+			}
+		} else { // else we are loading from the ssh/remote client
+			scpOpts.FromRemote = true
+			scpOpts.Connections = append(scpOpts.Connections, args[0])
+		}
+	} else if len(args) == 2 && strings.Contains(args[1], "::") && len(scpOpts.Connections) < 2 { // if we are given image host::
+		scpOpts.ToRemote = true
+		scpOpts.Connections = append(scpOpts.Connections, args[1])
+	}
+
+	for i, val := range scpOpts.Connections {
+		splitEnv := strings.SplitN(val, "::", 2)
+		switch len(splitEnv) {
+		case 1:
+			scpOpts.Connections[i] = splitEnv[0]
+		case 2:
+			if i == 0 {
+				scpOpts.ImageName = splitEnv[1]
+			} else if i == 1 {
+				scpOpts.Tag = splitEnv[1]
+			}
+			scpOpts.Connections[i] = splitEnv[0]
+		}
+
+		found := false
+		for name, data := range cfg.Engine.ServiceDestinations { // loop through all of the connections until we find ours
```

Comment:
This is a `map` after all, so just `conn, found := cfg.Engine.ServiceDestinations[scpOpts.Connections[i]]`, right?

Also in the other code path. (Should there be two?)

Reply:
I see what you are saying and I think that would work but it inhibits my ability to compare the connection names. I will try to implement this but might stick with what I have if it doesn't work

Reply:
Figured it out